src/status.rs

```diff
@@ -522,5 +560,63 @@ mod test {
         assert_eq!(status, deserialized);
     }
 
-}
+    #[test]
+    fn serialize_extension_fields_empty() {
+        let status = StatusBuilder::new("a")
+            .logo("b")
+            .url("c")
+            .location(Location::default())
+            .contact(Contact::default())
+            .build();
+        assert!(status.is_ok());
+        assert_eq!(
+            &to_string(&status.unwrap()).unwrap(),
+            "{\"api\":\"0.13\",\"space\":\"a\",\"logo\":\"b\",\"url\":\"c\",\
+            \"location\":{\"lat\":0.0,\"lon\":0.0},\"contact\":{},\"issue_report_channels\":[],\
+            \"state\":{\"open\":null}}"
+        );
+    }
+
+    #[test]
+    fn serialize_extension_fields() {
+        let status = StatusBuilder::new("a")
+            .logo("b")
+            .url("c")
+            .location(Location::default())
+            .contact(Contact::default())
+            .add_extension("aaa", Value::String("xxx".into()))
+            .add_extension("bbb", Value::Array(vec![Value::Null, Value::U64(42)]))
+            .build();
+        assert!(status.is_ok());
+        assert_eq!(
+            &to_string(&status.unwrap()).unwrap(),
+            "{\"api\":\"0.13\",\"space\":\"a\",\"logo\":\"b\",\"url\":\"c\",\
+            \"location\":{\"lat\":0.0,\"lon\":0.0},\"contact\":{},\"issue_report_channels\":[],\
+            \"state\":{\"open\":null},\"ext_aaa\":\"xxx\",\"ext_bbb\":[null,42]}"
+        );
+    }
 
+    /// Extension field names are automatically prepended with `ext_`.
+    /// If two extensions with the same name (before or after prepending) are
+    /// added, then the second call will overwrite the first one (standard Map
+    /// behavior).
+    #[test]
+    fn prepend_ext_to_extension_field_names() {
+        let status = StatusBuilder::new("a")
+            .logo("b")
+            .url("c")
+            .location(Location::default())
+            .contact(Contact::default())
+            .add_extension("aaa", Value::String("xxx".into()))
+            .add_extension("ext_aaa", Value::String("yyy".into()))
+            .add_extension("bbb", Value::Null)
+            .add_extension("ext_ccc", Value::Null)
+            .build();
+        assert!(status.is_ok());
+        let serialized = to_string(&status.unwrap()).unwrap();
+        assert!(serialized.contains("\"ext_aaa\":\"yyy\""));
+        assert!(serialized.contains("\"ext_bbb\":null"));
+        assert!(serialized.contains("\"ext_ccc\":null"));
```

Comment:
Maybe we should use https://serde.rs/unit-testing.html for our unit tests? But this will probably be in a separate pull request.